### PR TITLE
Healthbot rules for TCP and NSR

### PR DESCRIPTION
**Pull request by kssk92sathish**

### CODE REVIEW

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/*
+ * This rule checks for the TCP Embryonic connection drop count in the system.
+ * TCP Embryonic connection is a half-open connection.
+ *
+ * It shows YELLOW alarm when the count is increased by minimum 10 from its previous value.
+ * It's sampling time is 1 minute.
+ */
+iceberg {
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule]: Same as above. Please address

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    topic kernel.tcpip {
+        rule check-tcp-connection-drop-count {
+            synopsis "Detect TCP Embryonic connection drops";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule]: Same as above. Please address

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                    when {
+                        not-equal-to "$tcp-conndrops-cnt" 0;
+                        increasing-at-most-by-value "$tcp-conndrops-cnt" {
+                            value 9;
+                            time-range 140s;
+                        }
+                    }
+                    then {
+                        status {
+                            color green;
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule]: Same as above. Please address

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule [hunk 4]
```diff
@@ -0,0 +1,88 @@
+/*
+ * This rule checks for the TCP Distributed denial-of-service attack(DDoS) attack count in the system.
+ *
+ * It shows RED alarm when the count is increased by minimum 1 from its previous value.
+ * It's sampling time is 1 minute.
+ */
+iceberg {
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule]: Same as above. Please address

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                sensor ddos-attack-sensor {
+                    path tcp-ddos-attack;
+                }
+                type integer;
+                description "Indicates the system DDoS attack count";
+            }
+            trigger report-tcp-ddos-attack-count-iagent {
+                synopsis "Trigger for TCP DDoS Attack";
+                description "Checks for TCP DDoS Attack and raises an alarm";
+                frequency 1m;
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule]: Same as above. Please address

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule [hunk 6]
```diff
@@ -0,0 +1,87 @@
+/*
+ * This rule checks for the TCP Distributed denial-of-service attack(DDoS) attack count in the system.
+ *
+ * It shows RED alarm when the count is increased by minimum 1 from its previous value.
+ * It's sampling time is 1 minute.
+ */
+iceberg {
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule]: Same as above. Please address

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+                sensor ddos-attack-sensor {
+                    path /junos/kernel/tcpip/tcp/tcp-ddos-attack-cnt;
+                }
+                type integer;
+                description "Indicates the system DDoS attack count";
+            }
+            trigger report-tcp-ddos-attack-count {
+                synopsis "Trigger for TCP DDoS Attack";
+                description "Checks for TCP DDoS Attack and raises an alarm";
+                frequency 1m;
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule]: Same as above. Please address

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule [hunk 8]
```diff
@@ -0,0 +1,102 @@
+/*
+ * This rule checks for the TCP Embryonic connection drop count in the system.
+ * TCP Embryonic connection is a half-open connection.
+ *
+ * It shows YELLOW alarm when the count is increased by minimum 10 from its previous value.
+ * It's sampling time is 1 minute.
+ */
+iceberg {
```
[the remark this answers is not in the document]
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    topic kernel.tcpip {
+        rule check-tcp-connection-drop-count-iagent {
+            synopsis "Detect TCP Embryonic connection drops";
```
[the remark this answers is not in the document]
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+            description "Check whether the TCP connection are getting dropped";
+            sensor conn-drop-sensor {
+                synopsis "iAgent sensor for TCP Embryonic connection drops";
```
[the remark this answers is not in the document]
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+                sensor conn-drop-sensor {
+                    path tcp-connection-drops;
+                }
+                type integer;
+                description "Number of times the TCP connection got dropped";
+            }
+            trigger report-tcp-connection-drop-count-iagent {
+                synopsis "Trigger for TCP Embryonic connection drops";
+                description "Checks for TCP Embryonic connection drops and raises an alarm";
+                frequency 1m;
```
[the remark this answers is not in the document]
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule (hunk 1 above)
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule (hunk 2 above)
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule (hunk 4 above)
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule (hunk 5 above)
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count-iagent.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule (hunk 6 above)
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule (hunk 7 above)
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-ddos-attack-count.rule]: Fixed

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+                    when {
+                        not-equal-to "$tcp-conndrops-cnt" 0;
+                        increasing-at-most-by-value "$tcp-conndrops-cnt" {
+                            value 9;
+                            time-range 140s;
+                        }
+                    }
+                    then {
+                        status {
+                            color green;
```
[the remark this answers is not in the document]
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count-iagent.rule]: Color green is expected here. As part of PR 1460451 fix added "term no-tcp-connection-drops".

File: community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule (hunk 3 above)
Comment (kssk92sathish) [reply community_supplied/Kernel/kernel-tcpip/check-tcp-connection-drop-count.rule]: Color green is expected here. As part of PR 1460451 fix added "term no-tcp-connection-drops".

**Pull request merged**